backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -294,21 +296,114 @@ public void setUserIsActive(String username, Boolean active) {
+
+        LOG.info("Created Okta group={}", roleGroupName);
+
+        if (migration && MIGRATION_DEST_ROLES.contains(role)) {
+          migrationDestGroupIds.add(g.getId());
+          migrationDestGroupNames.add(roleGroupName);
+        }
+      } catch (ResourceException e) {
+        if (migration) {
+          // ignore attempts to create groups that already exist if this is in migration mode
+          migrationSourceGroupNames.add(roleGroupName);
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    for (Facility facility : facilities) {
+      String facilityGroupName = generateFacilityGroupName(externalId, facility.getInternalId());
+      String facilityGroupDescription =
+          generateFacilityGroupDescription(name, facility.getFacilityName());
+      try {
+        Group g =
+            GroupBuilder.instance()
+                .setName(facilityGroupName)
+                .setDescription(facilityGroupDescription)
+                .buildAndCreate(_client);
+        _app.createApplicationGroupAssignment(g.getId());
+
+        LOG.info("Created Okta group={}", facilityGroupName);
+
+      } catch (ResourceException e) {
+        if (migration) {
+          // ignore attempts to create groups that already exist if this is in migration mode
+          migrationSourceGroupNames.add(facilityGroupName);
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    if (migration && !migrationSourceGroupNames.isEmpty() && !migrationDestGroupIds.isEmpty()) {
+      String ruleName = generateMigrationGroupRuleName(externalId, MIGRATION_DEST_ROLES);
+      String ruleExpression = generateMigrationGroupRuleExpression(migrationSourceGroupNames);
+      try {
+        // https://developer.okta.com/docs/reference/api/groups/#group-rule-operations
+        GroupRule rule =
+            GroupRuleBuilder.instance()
+                .setType("group_rule")
+                .setName(ruleName)
+                .setGroupRuleExpressionType("urn:okta:expression:1.0")
+                .setGroupRuleExpressionValue(ruleExpression)
+                .setAssignUserToGroups(migrationDestGroupIds)
+                .buildAndCreate(_client);
+        rule.activate();
+
+        LOG.info(
+            "Migrated users from Okta source_groups={} to destination_groups={}",
+            migrationSourceGroupNames,
+            migrationDestGroupNames);
+      } catch (ResourceException e) {
+        LOG.error(
+            "Error migrating users from Okta source_groups={} to destination_groups={}: "
+                + "check if conflicting group_rule={} already exists. "
+                + "Did you delete groups without deleting their relevant group rules?",
+            migrationSourceGroupNames,
+            migrationDestGroupNames,
+            ruleName);
+      }
     }
 
     _app.update();
   }
 
+  public void createOrganization(Organization org) {
+    createOrganization(org, Set.of(), false);
+  }
+
```

Comment:
```suggestion
```
Interface takes care of this now, right?

Reply:
well that would seem to make sense, yes 😆